api/fake_agent_server_test.go

```diff
@@ -0,0 +1,154 @@
+package api_test
+
+import (
+	"context"
+	"log/slog"
+	"testing"
+
+	"github.com/buildkite/agent-stack-k8s/v2/api"
+	"github.com/buildkite/stacksapi"
+)
+
+func TestFakeAgentServer_DefaultBehavior(t *testing.T) {
+	ctx := context.Background()
+	
+	// Create fake server
+	server := api.NewFakeAgentServer()
+	defer server.Close()
+	
+	// Create REAL AgentClient pointing to fake server
+	client, err := api.NewAgentClient(ctx, api.AgentClientOpts{
+		Token:    "fake-token",
+		Endpoint: server.URL(),
+		StackID:  "test-stack",
+		Logger:   slog.Default(),
+	})
+	if err != nil {
+		t.Fatalf("NewAgentClient() error = %v", err)
+	}
+
+	jobIDs := []string{"job-1", "job-2", "job-3"}
+	result, retryAfter, err := client.ReserveJobs(ctx, jobIDs)
+
+	if err != nil {
+		t.Errorf("ReserveJobs() error = %v, want nil", err)
+	}
+	if retryAfter != 0 {
+		t.Errorf("ReserveJobs() retryAfter = %v, want 0", retryAfter)
+	}
+	if len(result.Reserved) != 3 {
+		t.Errorf("len(result.Reserved) = %d, want 3", len(result.Reserved))
+	}
+	if len(result.NotReserved) != 0 {
+		t.Errorf("len(result.NotReserved) = %d, want 0", len(result.NotReserved))
+	}
+}
+
+func TestFakeAgentServer_RecordsCalls(t *testing.T) {
+	ctx := context.Background()
+	
+	server := api.NewFakeAgentServer()
+	defer server.Close()
+	
+	client, err := api.NewAgentClient(ctx, api.AgentClientOpts{
+		Token:    "fake-token",
+		Endpoint: server.URL(),
+		StackID:  "test-stack",
+		Logger:   slog.Default(),
+	})
+	if err != nil {
+		t.Fatalf("NewAgentClient() error = %v", err)
+	}
+
+	_, _, err = client.ReserveJobs(ctx, []string{"job-1", "job-2"})
+	if err != nil {
+		t.Fatalf("ReserveJobs() error = %v", err)
+	}
+	_, _, err = client.ReserveJobs(ctx, []string{"job-3"})
+	if err != nil {
+		t.Fatalf("ReserveJobs() error = %v", err)
+	}
+
+	if got, want := server.ReserveCallCount(), 2; got != want {
+		t.Errorf("ReserveCallCount() = %d, want %d", got, want)
+	}
+
+	call0, err := server.GetReserveCallIDs(0)
+	if err != nil {
+		t.Fatalf("GetReserveCallIDs(0) error = %v", err)
+	}
+	if len(call0) != 2 {
+		t.Errorf("len(call0) = %d, want 2", len(call0))
+	}
+
+	call1, err := server.GetReserveCallIDs(1)
+	if err != nil {
+		t.Fatalf("GetReserveCallIDs(1) error = %v", err)
+	}
+	if len(call1) != 1 {
+		t.Errorf("len(call1) = %d, want 1", len(call1))
+	}
```

Comment:
Similarly, what about diffing `server.ReserveCalls`?

```suggestion
	want := [][]string{
		{"job-1", "job-2"},
		{"job-3"},
	}
	if diff := cmp.Diff(server.ReserveCalls, want); diff != "" {
		t.Errorf("after ReserveJobs, server.ReserveCalls diff (-got +want):\n%s", diff)
	}
```